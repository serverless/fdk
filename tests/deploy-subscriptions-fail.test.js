const fdk = require('../lib/index')
const eventGatewayProcesses = require('./event-gateway/processes')

const config = {
  functions: [
    {
      functionId: 'myFunctionOne',
      provider: {
        type: 'awslambda',
        arn: 'https://xxxxxxxxxx.execute-api.us-east-1.amazonaws.com/dev/test',
        region: 'us-east-1',
      },
    },
  ],
  subscriptions: [
    { functionId: 'myFunctionOne', event: 'pageVisited' },
    { functionId: 'myFunctionTwo', event: 'pageVisited' },
  ],
}

let eventGateway
let eventGatewayProcessId

beforeAll(() =>
  eventGatewayProcesses
    .spawn({
      configPort: 4025,
      apiPort: 4026,
    })
    .then(processInfo => {
      eventGatewayProcessId = processInfo.id
      eventGateway = fdk.eventGateway({
        url: `http://localhost:${processInfo.apiPort}`,
        configurationUrl: `http://localhost:${processInfo.configPort}`,
      })
    })
)

afterAll(() => {
  eventGatewayProcesses.shutDown(eventGatewayProcessId)
})

test('should fail to create multiple misconfigured subscriptions', () => {
  expect.assertions(1)
  return eventGateway.deploy(config).catch(err => {
    expect(err).toMatchSnapshot()
  })
})
