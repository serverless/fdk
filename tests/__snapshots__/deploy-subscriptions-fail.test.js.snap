// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should fail to create multiple misconfigured subscriptions 1`] = `[Error: Failed to configure the Event Gateway. Failed to configure the Event Gateway. Failed to subscribe the event pageVisited to the function myFunctionTwo due the error: Function "myFunctionTwo" not found.]`;
